### PR TITLE
Fix: Add checking for Uniqueness in usernames

### LINKED ISSUE #104: Fix: Add checking for Uniqueness in usernames
We want to be able to see if a username is already in use without having to catch the error

### PR DESCRIPTION
**Pull request by gregorysa**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/104)
- Endpoint to get a user by username
- Basic tests
- Incorporated into mobile login page and auth context as a function
- Called when the user clicks out of the text box
- Disable the submit button when the username is not unique

# How Has This Been Tested?
- Error state is set and button is disabled when the username is found in the database
- Clearing the text box re-enables the submit button and clears error state
- Typing a unique username enables the submit button and clears error state

## Screenshots
https://github.com/user-attachments/assets/43072c71-4bd7-49c3-bacb-fe7022db2568

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

**Comment by gregorysa** [comment]
the endpoint might need to be changed to return just a boolean instead of the user, and i'm not sure how this endpoint will work with the middleware

**Comment by adescoteaux1** [comment]
> the endpoint might need to be changed to return just a boolean instead of the user, and i'm not sure how this endpoint will work with the middleware

fixed

File: backend/internal/models/user.go [hunk 1]
```diff
@@ -68,3 +68,14 @@ type DeleteUserInput struct {
 type DeleteUserOutput struct {
 	Body *User `json:"body"`
 }
+
+// Username exists check
+type GetUserByUsernameInput struct {
```
Comment (josh-torre) [inline backend/internal/models/user.go]: Nit: Throughout you have "Get user" but you also have "user exists" can you go through your changes and make this consistent?

**Comment by josh-torre** [comment]
@gregorysa Also please make sure the FE is formatted and resolve the merge conflicts :pray:

**Comment by adescoteaux1** [comment]
@gregorysa let us know when this is ready for re-review pls

**Review by josh-torre — APPROVED** [review]
